js_src/_dart_helpers.js

```diff
@@ -127,7 +127,9 @@ function _createReactDartComponentClass2(dartInteropStatics, componentStatics, j
       dartInteropStatics.handleComponentDidCatch(this.dartComponent, error, info);
     }
     static getDerivedStateFromError(error) {
-      return dartInteropStatics.handleGetDerivedStateFromError(componentStatics, error);
+      var result = dartInteropStatics.handleGetDerivedStateFromError(componentStatics, error);
+      if (typeof result === 'undefined') result = null;
+      return result;
```

Comment:
#nit I prefer the use of `let` as seen in `getDerivedStateFromProps` above...

```js
let derivedState = dartInteropStatics.handleGetDerivedStateFromError(componentStatics, error);
return typeof derivedState !== 'undefined' ? derivedState : null;
```

Reply:
fair enough